src/main/java/com/suse/saltstack/netapi/parser/JsonParser.java

```diff
@@ -92,18 +92,22 @@ public T parse(InputStream inputStream) {
     }
 
     /**
-     * Deserializer for date representation received from the API
+     * TypeAdapter for date representation received from the API
      * (which represents it as a (floating) number of seconds since the Epoch).
      */
-    private class SaltStackDateDeserializer implements JsonDeserializer<Date> {
+    private class DateAdapter extends TypeAdapter<Date> {
+
         @Override
-        public Date deserialize(JsonElement jsonElement, Type type,
-                JsonDeserializationContext jsonDeserializationContext)
-                throws JsonParseException {
+        public void write(JsonWriter jsonWriter, Date date) throws IOException {
+            throw new UnsupportedOperationException("Writing JSON not supported.");
+        }
+
+        @Override
+        public Date read(JsonReader jsonReader) throws IOException {
             try {
-                double dateMiliSecs = jsonElement.getAsDouble() * 1000;
+                double dateMiliSecs = jsonReader.nextDouble() * 1000;
```

Comment:
Very minor nitpick: milliseconds is written with double "l". I would suggest to use `dateMilliseconds` as the name for this variable (milliseconds is only one word).